src/OrchardCore.Modules/OrchardCore.Workflows/Services/SignalService.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using Microsoft.AspNetCore.DataProtection;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace OrchardCore.Workflows.Services
+{
+    public class SignalService : ISignalService
```

Comment:
This will be reviewed in details, because we need to be sure it's time sensitive, and that it handles different machines.
And I don't like the name ;) (which I know is coming from O1, and I probably chose it)

Reply:
And isolated per tenant

Reply:
Since this will end up using the same pattern already leveraged in the Users and OpenID modules, I suggest introducing a new `OrchardCore.DataProtection` module to take care of injecting a tenant-aware `IDataProtectionProvider` in the modules `IServiceCollection`.

Reply:
I agree with @PinpointTownes' suggestion - the functionality is very generic, and if it's also used in Users and OpenID then it makes even more sense. If @sebastienros agrees, then I'll create a separate PR doing that work, then once that's merged into dev, I'll merge dev into workflows and get rid of this SignalService class in favor of the new DataProtection package.

Reply:
Not a module, but a project where we can do `.AddDataProtectors()`, that can be called by the modules that need it (it would only register once). We need to think about how to get these data protectors, some need to work across tenants (and not), some need to work across machines (and not) and also timed. Probably need to work on it in a different PR.

Reply:
@sfmskywalker @sebastienros I opened https://github.com/OrchardCMS/OrchardCore/issues/1391 to track that.